### PR TITLE
test: add usecase test interceptor

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Each test could be run with `-Dusecase.test.repo.dir=<path_for_testcase_source>`, this would generate a subfolder folder under provided path_for_testcase_source with following structure and place cobol source(`source.cbl`), copybooks(files with `.cpy extensnsion` and a marker files indicating if Cobol LS engine expects 0 diagnostics)
      ` class name --> test method name --> parameter number (if present) `
      
      For example:  `mvn surefire:test -pl engine -Dtest=Test88ThruHover -Dusecase.test.repo.dir=<path>`
      

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/suite/UniqueIdTestWatcher.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            System.out.println("Created file: " + filePath);
+          } catch (IOException e) {
+            System.err.println("Error writing copybook: " + e.getMessage());
+          }
+        });
+  }
+
+  private static void writeTestData(TestData testData, Path mainFolderPath) throws IOException {
+    Path testDataPath = mainFolderPath.resolve("testData.txt");
+    Files.deleteIfExists(testDataPath);
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/suite/UniqueIdTestWatcher.java]: I don't think these conditional deletions should be here... The code overall does not handle removal of copybooks, or NoTestData file etc.

File: server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * ExtensionContextCleaner
+ */
+public class ExtensionContextCleaner implements AfterEachCallback {
+
+    /**
+     *
+     * @param context
+     */
+    @Override
+    public void afterEach(ExtensionContext context) {
```
Comment (slavek-kucera) [inline server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java]: What is the reason to keep this one method separate?

File: server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java [hunk 3]
```diff
@@ -278,6 +280,7 @@ public AnalysisResult runTest(
                 .build(),
             languageId);
     assertResultEquals(actual, document.getTestData());
+    UseCaseUtils.storeDocumentToUnitTextExtensionContext(document.getText(), document.getCopybooks(), document.getTestData());
```
Comment (slavek-kucera) [inline server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java:283]: Do I understand correctly that the preprocessed code will be stored in the output?

Comment (slavek-kucera) [inline server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java:283]: Isn't this fuction called by `analyze`?

File: server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java (hunk 2 above)
Comment (amanPras) [reply server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java]: I will place this under ExtensionContextProvider, or maybe place everything inside UniqueIdTestWatcher

File: server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java (hunk 3 above)
Comment (amanPras) [reply server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java:283]: Yes

Comment (amanPras) [reply server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java:283]: Yes.
There are test which call analyze and don't provide any assertions to the usecase engine ( e.g. TestOutlineTreeLineNumbers) . So it has null as the testdata, To handle this I placed this method at 2 places,
one at analyze and once here. Calling it here updates the hashmap with the testData

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/suite/UniqueIdTestWatcher.java (hunk 1 above)
Comment (amanPras) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/suite/UniqueIdTestWatcher.java]: updated.

File: server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java (hunk 2 above)
Comment (amanPras) [reply server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/ExtensionContextCleaner.java]: Placed it under ExtensionContextProvider. As this is  required only in test module

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**